src/index.ts

```diff
@@ -1006,12 +1039,36 @@ class Envoy {
     return resp;
   }
 
+  async getServiceLoggerCellId(hha_hash) {
```

Comment:
I like the extraction of this for general reuse.  - Are we using this in more than one place though?  I saw your comment that it breaks in `logServiceConfirmation`, but do you have any leads on why?

Reply:
No. I did some investigation but couldn't figure it out. It didn't make sense to me. So currently we're just using it in one place. I think it's still worth extracting it though as it makes sense as a small self-contained function.